### ISSUE TITLE
Patch2self denoising followed by topup and eddy corrections worsens distortions in the orbitofrontal region

### ISSUE THREAD

**Issue by jayashree15** [dipy/dipy#2812]
## Ran Patch2self followed by eddy and topup which seem to worsen the distortion in 7T diffusion brain dataset
Data - 7T (Terra), b = 1500, dir = 54, PE = AP/PA

Attached is the screenshot of the eddy/topup output dataset (b0 volume) with and without running patch2self before eddy/topup. 

![Screen Shot 2023-05-17 at 2 30 24 PM](https://github.com/dipy/dipy/assets/44578765/3c4eff84-6bc4-4fc6-98c8-0abc22a0a504)
![Screen Shot 2023-05-17 at 2 31 33 PM](https://github.com/dipy/dipy/assets/44578765/85a3a009-883c-4817-b9aa-4c3c923253f0)

Ran P2S with DIPY 1.7.0 (ran with default parameters with just diffusion volume and b-value as inputs) and eddy_cuda9.1

Please let me know if there are any parameters that can be changed with P2S before eddy/topup. 

Thank you,

Jayashree

**Comment by ShreyasFadnavis** [comment]
Hi @jayashree15 ! Thank you for using Patch2Self 😃 

Here are a couple of sanity checks:

1. If running after TOPUP, why are there distortions in the data? 

2. Can you try re-running, using the option: 'b0_denoising=False'. This will just switch off the b0 denoising. It maybe the case that there are some bad b0 volumes causing this issue. Just flipping through the b0s should help understand the issue.


Do you also see this issue in the non-b0 volumes?

Thanks,
Shreyas

**Comment by jayashree15** [comment]
Thank you, Shreyas. 

I ran P2S first before doing eddy/topup. Somehow, P2S applied data seem to do worse when apply eddy/topup and yes I do see it in the non-b0 volumes too. Attached is a shot from non b0 volume.

![Screen Shot 2023-05-17 at 5 06 06 PM](https://github.com/dipy/dipy/assets/44578765/3a3844bc-f78a-4709-bc65-336818538b78)
![Screen Shot 2023-05-17 at 5 06 17 PM](https://github.com/dipy/dipy/assets/44578765/2fcabc40-a856-4d8a-abf1-04817414b201)

I also tried denoising using other method such as NORDIC and I don't encounter worsening of susceptibility artifact due to denoising. 

Do you think I could try denoising after doing eddy and topup correction?

Jayashree

**Comment by ShreyasFadnavis** [comment]
Yes please, that was going to be my suggestion -- run Eddy/TOPUP and then Patch2Self.

**Comment by jayashree15** [comment]
Ok, that helped with minimizing the distortion. I'm curious why you recommended doing the denoising before the eddy/topup correction during the workshop and by not following what implications might follow for us?

On another note - I saw a screenshot where you had combined NORDIC and P2S denoising methods on some dataset. Did you find anything interesting? Would you be interested in connecting regarding applying P2S and/other denoising algorithm on our dataset? If yes, please let me know.

Thank you!

**Comment by ShreyasFadnavis** [comment]
@jayashree15 100%! Let me know how I can help with this :)
